controller-server/src/main/java/com/yahoo/vespa/hosted/controller/ApplicationController.java

```diff
@@ -338,37 +330,13 @@ public ActivateResult deploy(ApplicationId applicationId, ZoneId zone,
                 // TODO(jvenstad): Remove this when all packages are validated upon submission, as in ApplicationApiHandler.submit(...).
                 verifyApplicationIdentityConfiguration(applicationId.tenant(), applicationPackage, deployingIdentity);
 
-                // Assign global rotation
-                // TODO(ogronnesby): Remove feature flag and replace calls to withRotationLegacy with withRotation
-                // TODO(mpolden): Remove all handling of legacy endpoints once withRotationLegacy disappears
-                if (useMultipleEndpoints.with(FetchVector.Dimension.APPLICATION_ID, application.get().id().serializedForm()).value()) {
-                    application = withRotation(application, zone);
-
-                    // Include global DNS names
-                    Application app = application.get();
-                    app.rotations().stream()
-                       .filter(assignedRotation -> assignedRotation.regions().contains(zone.region()))
-                       .map(assignedRotation -> {
-                                return new ContainerEndpoint(
-                                        assignedRotation.clusterId().value(),
-                                        Stream.concat(
-                                                app.endpointsIn(controller.system(), assignedRotation.endpointId()).legacy(false).asList().stream().map(Endpoint::dnsName),
-                                                Stream.of(assignedRotation.rotationId().asString())
-                                        ).collect(Collectors.toList())
-                                );
-                            })
-                       .forEach(endpoints::add);
-                } else {
-                    application = withRotationLegacy(application, zone);
-
-                    // Add both the names we have in DNS for each endpoint as well as name of the rotation so healthchecks works
-                    Application app = application.get();
-                    app.endpointsIn(controller.system()).asList().stream().map(Endpoint::dnsName).forEach(legacyRotations::add);
-                    app.rotations().stream()
-                       .map(AssignedRotation::rotationId)
-                       .map(RotationId::asString)
-                       .forEach(legacyRotations::add);
-                }
+                // Assign and register endpoints
+                application = withRotation(application, zone);
+                endpoints = registerEndpointsInDns(application.get(), zone);
+                legacyRotations = endpoints.stream()
+                                           .map(ContainerEndpoint::names)
+                                           .flatMap(Collection::stream)
+                                           .collect(Collectors.toSet());
```

Comment:
Where is the promised `.editorconfig` file? 😛

Reply:
Real soon now 😆